### ISSUE TITLE
[otbn, sec_cm] Predecoded MAC Enable does not match Expected MAC enable

### ISSUE THREAD

**Issue by ctopal** [lowRISC/opentitan#13236]
The issue comes from the assignment of `mac_bignum_en_o` at `otbn_controller`: 
https://github.com/lowRISC/opentitan/blob/5948fabb7add084eacddc779fe275872cd7736f1/hw/ip/otbn/rtl/otbn_controller.sv#L915

We are using that output as enable of the `otbn_mac_bignum` module and then set our `expected_op_en` and `expected_acc_rd_en` with that. See: https://github.com/lowRISC/opentitan/blob/5948fabb7add084eacddc779fe275872cd7736f1/hw/ip/otbn/rtl/otbn_mac_bignum.sv#L231-L232

That causes a mismatch when an invalid instruction comes through resulting with an incorrect behaviour (since the idea is to make decoding redundant). I believe we should either change the relevant `otbn_predecode` section to also include `insn_valid_i` here: https://github.com/lowRISC/opentitan/blob/5948fabb7add084eacddc779fe275872cd7736f1/hw/ip/otbn/rtl/otbn_predecode.sv#L431-L432

or change the relevant part in the controller to not involve `insn_valid_i`.

What do you think?

**Comment by ctopal** [comment]
I think we are blanking if the predecoded enable signals are low. In the case of instruction not being valid, I thought we would also want to blank the internals. This came up when I was trying to implement some assertions around the functionality so if you think it doesn't make sense it's totally fine I can drop the assertions :)

**Comment by andreaskurth** [comment]
I did not think about this from the blanking perspective, sorry :)

As we just discussed, blanking is currently not designed to be mandatory for illegal instructions. The rationale is that blanking hinders SCA, and SCA primarily targets correct programs, where there are no illegal instructions. If an attacker were to combine SCA with FI to disable blanking through illegal instructions, OTBN would generate a recoverable error due to the illegal instruction. This would result in a secure wipe, so the algorithm during which the fault was injected would not terminate. Also, if too many recoverable errors happen in a given time frame, OTBN may escalate into a fatal error. We thus concluded that not blanking for illegal instructions is not a problem.
